src/store/index.js

```diff
@@ -33,6 +46,28 @@ export default new Vuex.Store({
     async setMovieTitle({ commit }, title) {
       commit("setMovieTitle", title);
     },
+    async login({ commit, dispatch }, credentials) {
+      await dispatch("logout");
+      const response = await authApi.login(credentials);
+      const authHeader = response.headers["authorization"];
+      setAuthHeader(authHeader);
+      commit("setAuthHeader", { authHeader });
+    },
+    async logout({ commit, getters }) {
+      if (!getters.isLoggedIn) return;
+      removeAuthHeader();
+      commit("resetAuthHeader");
+    },
+    restoreAuthSession({ commit }) {
+      const authHeader = localStorage.getItem(AUTH_HEADER_STORAGE_KEY);
+      if (authHeader) {
+        setAuthHeader(authHeader);
+        commit("setAuthHeader", { authHeader });
+      }
+    },
```

Comment:
We could also do all this when defining the default state of our application 💡.
This way we would not need this action and we would also not need to call it!

So in your state instead of:

```js
state: {
   // ...,
   authHeader: null
}
```

you could do this:

```js
state: {
   // ...,
   authHeader: localStorage.getItem(AUTH_HEADER_STORAGE_KEY)
}
```

This way we have the `authHeader` restored right off the bat~! 🚀

Reply:
>  This way we would not need this action and we would also not need to call it!

very much true for some cases but don't forget about `setAuthHeader` 😉